spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -338,6 +338,9 @@ protected void doStop() {
 	protected void actualStart() throws Exception {
 		this.aborted = false;
 		this.hasStopped = false;
+		if (getPrefetchCount() < this.messagesPerAck) {
+			setPrefetchCount(this.messagesPerAck);
```

Comment:
Good, thanks.
I fully understand the "smaller". I'm only concerted that with "bigger" we don't let other concurrent consumers to consume those extra messages. But yeah, that is already up to end-user. To have keep unacked messages for the next fetch request, or distribute all the message between concurrent consumers making `prefetch = messagesPerAck`.

Merging...